src/components/vanilla/controls/DatePicker/components/DateRangeWithGranularity.tsx

```diff
@@ -0,0 +1,147 @@
+import { dateParser } from '@cubejs-backend/api-gateway/dist/src/dateParser.js';
+import { DataResponse, Dimension, Granularity, TimeRange } from '@embeddable.com/core';
+import React, { useCallback, useEffect, useLayoutEffect, useMemo, useRef, useState } from 'react';
+import DateRangePicker from './DateRangePicker';
+import Dropdown from '../../Dropdown';
+import { getValidGranularities, getNote, getComparisonOptions, getComparisonPeriod } from '../utils/dateUtils'
+
+const valueProp: Dimension = {
+  __type__: 'dimension',
+  name: 'value',
+  nativeType: 'string',
+  title: 'Value'
+};
+
+export type Props = {
+  defaultPeriod?: TimeRange;
+  defaultComparison?: string;
+  defaultGranularity?: Granularity;
+  showGranularity?: boolean;
+  onChangePeriod: (v: TimeRange | null) => void;
+  onChangeComparison: (v: TimeRange | null) => void;
+  onChangeGranularity: (v: Granularity | null) => void;
+};
+
+export default function DateRangeWithGranularity(props: Props) {
+  const { onChangeComparison, onChangePeriod, defaultPeriod } = props;
+
+  const ref = useRef<HTMLDivElement | null>(null);
+  const [hideDate, setHideDate] = useState(false);
+  const [period, setPeriod] = useState(defaultPeriod);
+  const [granularity, setGranularity] = useState(props.defaultGranularity);
+  const [compareOption, setCompareOption] = useState(props.defaultComparison);
+
+  useLayoutEffect(() => {
+    const interval = setInterval(() => {
+      const width = ref.current?.clientWidth;
+
+      if (width) setHideDate(width < 250);
+    }, 100);
+
+    return () => clearInterval(interval);
+  }, []);
+
+  const granularityOptions = getValidGranularities(period);
+
+  const comparisonOptions: DataResponse = useMemo(() => {
+    return {
+      isLoading: false,
+      data: getComparisonOptions(period)
+    };
+  }, [period?.from, period?.to]);
```

Comment:
there is a missing dependency according to git:

<img width="455" alt="image" src="https://github.com/user-attachments/assets/1c4b6656-68da-45b4-a7c3-e1f80407c9e4">


use the period, instead of period.from, period.to